app/Http/Controllers/RentalController.php

```diff
@@ -2,63 +2,75 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Rental;
+use App\Repositories\Contracts\RentalRepositoryInterface;
 use Illuminate\Http\Request;
 
 class RentalController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     */
-    public function index()
+    protected $rentalRepository;
+
+    public function __construct(RentalRepositoryInterface $rentalRepository)
     {
-        //
+        $this->rentalRepository = $rentalRepository;
     }
 
-    /**
-     * Show the form for creating a new resource.
-     */
-    public function create()
+    public function index()
     {
-        //
+        return response()->json($this->rentalRepository->all());
     }
 
-    /**
-     * Store a newly created resource in storage.
-     */
     public function store(Request $request)
     {
-        //
-    }
+        $validatedData = $request->validate([
+            'skate_park_id' => 'required|exists:skate_parks,id',
+            'renter_name' => 'required|string|max:255',
+            'start_time' => 'required|date',
+            'end_time' => 'required|date|after:start_time',
+        ]);
```

Comment:
```suggestion
```

em `public function store(Request $request)` troque `Request` por uma classe que extenda o `Request`

Dentro dessa classe que extende Request (ex.: StoreRentalRequest.php) deve haver um método rules() que retorna um array.

Esse array de rules pode ser identico a essas linhas, assim, no momento que a requisição for feita, o Laravel automaticamente irá validar os dados de request com as regras da sua classe (abstraindo responsabilidade do Controller para o Request)

Reply:
Exemplo de classe:

```
<?php

namespace App\Api\Product\Requests;

use Illuminate\Http\Request;

class StoreRequestExample extends Request
{
    public function rules(): array
    {
        return [
            'skate_park_id' => 'required|exists:skate_parks,id',
            'renter_name' => 'required|string|max:255',
            'start_time' => 'required|date',
            'end_time' => 'required|date|after:start_time',
        ];
    }
}
```